src/main/java/uk/gov/pay/ledger/transaction/dao/TransactionDao.java

```diff
@@ -71,27 +72,36 @@ public TransactionDao(Jdbi jdbi) {
     }
 
     public List<TransactionEntity> searchTransactions(TransactionSearchParams searchParams) {
-        String searchExtraFields = searchParams.generateQuery();
         return jdbi.withHandle(handle -> {
-            Query query = handle.createQuery(SEARCH_QUERY_STRING.replace(":searchExtraFields", searchExtraFields));
-            searchParams.getQueryMap().forEach(query::bind);
+            Query query = getQuery(searchParams, handle, SEARCH_QUERY_STRING);
             return query
                     .map(new TransactionMapper())
                     .list();
         });
     }
 
     public Long getTotalForSearch(TransactionSearchParams searchParams) {
-        String searchExtraFields = searchParams.generateQuery();
         return jdbi.withHandle(handle -> {
-            Query query = handle.createQuery(SEARCH_COUNT_QUERY_STRING.replace(":searchExtraFields", searchExtraFields));
-            searchParams.getQueryMap().forEach(query::bind);
+            Query query = getQuery(searchParams, handle, SEARCH_COUNT_QUERY_STRING);
             return query
                     .mapTo(Long.class)
                     .findOnly();
         });
     }
 
+    private Query getQuery(TransactionSearchParams searchParams, Handle handle, String searchCountQueryString) {
+        String searchExtraFields = searchParams.generateQuery();
+        Query query = handle.createQuery(searchCountQueryString.replace(":searchExtraFields", searchExtraFields));
+        searchParams.getQueryMap().forEach((k, v) -> {
```

Comment:
This took me a minute to wrap my head around!

Even though it's more verbose maybe something like `searchKey` and `searchValue` would help with understanding what this is doing. 

Or moving it to a static named method that's passed in to `forEach`, this is probably simple enough to not need it.

Reply:
I see. Thanks. Will have a look at it separately.